src/main/java/com/openshift/jenkins/plugins/util/ClientCommandRunner.java

```diff
@@ -95,109 +99,60 @@ public Boolean call() throws IOException, InterruptedException {
-             InputStream redirectedStdout = new FastPipedInputStream(stdout);
+             FastPipedInputStream redirectedStdout = new FastPipedInputStream(stdout);
              FastPipedOutputStream stderr = new FastPipedOutputStream();
-             InputStream redirectedStderr = new FastPipedInputStream(stderr)) {
+             FastPipedInputStream redirectedStderr = new FastPipedInputStream(stderr)) {
             // running `oc` remotely
-            // NOTE: Launcher is not used to start a remote process because of Jenkins bugs described on https://issues.jenkins-ci.org/browse/JENKINS-53586
-            //   and https://issues.jenkins-ci.org/browse/JENKINS-53422.
-            remoteOCProcessFuture = filePath.actAsync(new OcCallable(command, envVars, new RemoteOutputStream(stdout), new RemoteOutputStream(stderr)));
-
-            // reading the output (stdout and stderr) from the remote `oc` process
-            CompletionService<Boolean> completionService = new ExecutorCompletionService<>(pool);
+            Launcher.ProcStarter ps = launcher.launch().cmds(Arrays.asList(command)).envs(envVars).pwd(filePath).quiet(true).stdout(stdout).stderr(stderr);
 
             // handling stderr
-            ocOutputConsumerFutures.add(completionService.submit(new OcOutputConsumer(redirectedStderr, stderrOutputObserver)));
+            futures.add(completionService.submit(new OcOutputConsumer(redirectedStderr, stderrOutputObserver)));
 
             // handling stdout
-            ocOutputConsumerFutures.add(completionService.submit(new OcOutputConsumer(redirectedStdout, stdoutOutputObserver)));
-
-            // waiting for output handlers to stop
-            for (int i = 0; i < ocOutputConsumerFutures.size(); ++i) {
-                boolean shouldInterrupt = completionService.take().get();
-                if (shouldInterrupt) { // an observer requests interrupting the remote `oc` process
-                    remoteOCProcessFuture.cancel(true);
+            futures.add(completionService.submit(new OcOutputConsumer(redirectedStdout, stdoutOutputObserver)));
+
+            // start remote `oc` process
+            proc = ps.start();
+            final Proc proc1 = proc; // make it "final"
+            // future to wait for remote `oc` process to stop
+            Future<Object> procFuture = completionService.submit(() -> proc1.join());
+            futures.add(procFuture);
+
+            // waiting for any of oc process, stdout consumer, and stderr consumer to stop
+            for (int i = 0; i < futures.size(); ++i) {
+                Future<Object> completedFuture = completionService.take();
+                if (completedFuture == procFuture) {
+                    // the remote `oc` process has exited
+                    exitStatus = (int) completedFuture.get();
+                    // closing stdout/stderr so that observers will stop
+                    stdout.close();
+                    stderr.close();
```

Comment:
close the redirected streams as well .... let's not assume closing the inner stream closes those